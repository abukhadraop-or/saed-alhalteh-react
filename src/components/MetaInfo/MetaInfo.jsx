import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import {
  MetaInfoContainer,
  ProfileImg,
  Username,
  PublishDate,
  NameAndDateContainer,
} from './meta-info.styles';

/**
 * Component that shows the post's author username and image, and the date when the posts was created.
 *
 * @param {Object} props The props object.
 * @param {Object} props.author Object contains details about the post's author.
 * @param {String} props.createdAt The date when the post created.
 *
 * @return {JSX.Element} MetaInfo component.
 */
function MetaInfo({ author, createdAt }) {
  const date = Date.parse(createdAt);
  const dateObj = new Date(date);
  const printDate = dateObj.toDateString();

  return (
    <>
      <MetaInfoContainer>
        <Link to={{ pathname: '/user', state: { author } }}>
          <ProfileImg src={author.image} alt="profile" />
        </Link>
        <NameAndDateContainer>
          <Username to={{ pathname: '/user', state: { author } }}>{author.username}</Username>
          <PublishDate>{printDate}</PublishDate>
        </NameAndDateContainer>
      </MetaInfoContainer>
    </>
  );
}

MetaInfo.propTypes = {
  author: PropTypes.shape({
    username: PropTypes.string,
    image: PropTypes.string,
  }).isRequired,
  createdAt: PropTypes.string.isRequired,
};

export default MetaInfo;
