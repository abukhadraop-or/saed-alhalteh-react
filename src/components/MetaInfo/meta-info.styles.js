import { Link } from 'react-router-dom';
import styled from 'styled-components';
import { colors } from 'styles/colors-and-breaking-points';

export const MetaInfoContainer = styled.div`
  display: flex;
`;

export const ProfileImg = styled.img`
  align-self: center;
  border-radius: 50%;
  cursor: pointer;
  height: 2.5rem;
  width: 2.5rem;
`;

export const NameAndDateContainer = styled.div`
  align-self: center;
  padding: 0 0 0 0.3125rem;
`;

export const Username = styled(Link)`
  color: ${colors.fern};
  font-size: 0.875rem;
  text-decoration: none;
  &:hover {
    color: ${colors.goblin};
    text-decoration: underline;
  }
`;

export const PublishDate = styled.div`
  color: ${colors.oFern};
  font-size: 0.75rem;
  margin: 0;
`;
