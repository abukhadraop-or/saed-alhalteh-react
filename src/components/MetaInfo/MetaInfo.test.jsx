import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import MetaInfo from '.';

const metaInfoProps = {
  author: {
    username: 'Jack',
    image:
      'https://upload.wikimedia.org/wikipedia/en/thumb/2/24/Crystal_personal.svg/1024px-Crystal_personal.svg.png',
  },
  createdAt: '2016-02-18T03:22:56.637Z',
};

describe('<MetaInfo/> tests', () => {
  it('matches the snapshot', () => {
    const wrapper = shallow(
      <MetaInfo author={metaInfoProps.author} createdAt={metaInfoProps.createdAt} />
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
