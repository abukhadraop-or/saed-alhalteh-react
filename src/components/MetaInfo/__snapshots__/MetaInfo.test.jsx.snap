// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MetaInfo/> tests matches the snapshot 1`] = `
<Fragment>
  <styled.div>
    <Link
      to={
        Object {
          "pathname": "/user",
          "state": Object {
            "author": Object {
              "image": "https://upload.wikimedia.org/wikipedia/en/thumb/2/24/Crystal_personal.svg/1024px-Crystal_personal.svg.png",
              "username": "Jack",
            },
          },
        }
      }
    >
      <styled.img
        alt="profile"
        src="https://upload.wikimedia.org/wikipedia/en/thumb/2/24/Crystal_personal.svg/1024px-Crystal_personal.svg.png"
      />
    </Link>
    <styled.div>
      <Styled(Link)
        to={
          Object {
            "pathname": "/user",
            "state": Object {
              "author": Object {
                "image": "https://upload.wikimedia.org/wikipedia/en/thumb/2/24/Crystal_personal.svg/1024px-Crystal_personal.svg.png",
                "username": "Jack",
              },
            },
          }
        }
      >
        Jack
      </Styled(Link)>
      <styled.div>
        Thu Feb 18 2016
      </styled.div>
    </styled.div>
  </styled.div>
</Fragment>
`;
